java/adapter/orc/src/main/java/org/apache/arrow/adapter/orc/OrcStripeReader.java

```diff
@@ -0,0 +1,157 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.adapter.orc;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import org.apache.arrow.flatbuf.MessageHeader;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.arrow.vector.ipc.ReadChannel;
+import org.apache.arrow.vector.ipc.message.ArrowDictionaryBatch;
+import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
+import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+import org.apache.arrow.vector.ipc.message.MessageChannelReader;
+import org.apache.arrow.vector.ipc.message.MessageResult;
+import org.apache.arrow.vector.ipc.message.MessageSerializer;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.arrow.vector.util.ByteArrayReadableSeekableByteChannel;
+
+import io.netty.buffer.ArrowBuf;
+
+
+public class OrcStripeReader extends ArrowReader {
+  private OrcStripeReaderJniWrapper stripeReader;
+  private MessageChannelReader schemaReader;
+
+  protected OrcStripeReader(OrcStripeReaderJniWrapper stripeReader, BufferAllocator allocator) {
+    super(allocator);
+    this.stripeReader = stripeReader;
+  }
+
+  /**
+   * Load the next ArrowRecordBatch to the vector schema root if available.
+   *
+   * @return true if a batch was read, false on EOS
+   * @throws IOException on error
+   */
+  @Override
+  public boolean loadNextBatch() throws IOException {
+    OrcRecordBatch recordBatch = stripeReader.next();
+    if (recordBatch == null) {
+      return false;
+    }
+
+    ArrayList<ArrowBuf> buffers = new ArrayList<>();
+    for (OrcMemoryJniWrapper buffer : recordBatch.buffers) {
+      buffers.add(new ArrowBuf(
+              new OrcReferenceManager(buffer),
+              null,
+              (int)buffer.getSize(),
+              buffer.getMemoryAddress(),
+              false));
+    }
+
+    loadRecordBatch(new ArrowRecordBatch(
+            recordBatch.length,
+            recordBatch.nodes.stream()
+              .map(buf -> new ArrowFieldNode(buf.getLength(), buf.getNullCount()))
+              .collect(Collectors.toList()),
+            buffers));
+    return true;
+  }
+
+  /**
+   * Return the number of bytes read from the ReadChannel.
+   *
+   * @return number of bytes read
+   */
+  @Override
+  public long bytesRead() {
+    return 0;
```

Comment:
is this intentional

Reply:
yes, this is intentional cause OrcReader doesn't provide corresponding bytesRead interface for accessing bytes actually read.

Reply:
maybe document it is as such?  Is it hard to expose?  @praveenbingo is having this accurate important?